lib/synapse/service_watcher/base.rb

```diff
@@ -33,16 +33,10 @@ def initialize(opts={}, synapse)
       @leader_election = opts['leader_election'] || false
       @leader_last_warn = Time.now - LEADER_WARN_INTERVAL
 
-      # the haproxy config
-      @haproxy = opts['haproxy']
-      @haproxy['server_options'] ||= ""
-      @haproxy['server_port_override'] ||= nil
-      %w{backend frontend listen}.each do |sec|
-        @haproxy[sec] ||= []
-      end
-
-      unless @haproxy.include?('port')
-        log.warn "synapse: service #{name}: haproxy config does not include a port; only backend sections for the service will be created; you must move traffic there manually using configuration in `extra_sections`"
+      @generator_config = {}
+      @synapse.available_generators.each do |generator_name, generator|
+        @generator_config[generator_name] = opts[generator_name] || {}
```

Comment:
Ok, I've made it more functional but I didn't want to make it a separate function because I don't really want to persist opts. Punt for now?